### PR TITLE
Add training modules from other initiatives

### PR DESCRIPTION
**Pull request by klieret**
The added modules are of somewhat different format, but they cover blindspots that the HSF modules have. 

Also it marks the reana training as stable (we've had a student walk through all of them and he didn't have issues)

### CODE REVIEW

File: _data/training-modules.yaml [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+  videos: ''
+  status: beta
+- description: 'This course covers various best practices like testing, pytest, object oriented programming, packing, CI, and more.'
+  name: "Software Engineering for Scientific Computing"
+  repository: https://github.com/henryiii/se-for-sci
+  status: stable
+  videos: ''
+  webpage: https://henryiii.github.io/se-for-sci/
+  id: se-for-sci
 - description: 'Track code changes, undo mistakes, collaborate. This module is a must.'
```
Comment (wdconinc) [inline _data/training-modules.yaml:15]:
```suggestion
 - description: 'Track code changes, undo mistakes, collaborate. This module is a must.'
```
This is more than a module and overlaps with many of our other modules. I'd avoid calling it a 'must' unless you specify something like 'for project maintainers' or so.

**Review by wdconinc — APPROVED** [review]

File: _data/training-modules.yaml (hunk 1 above)
Comment (klieret) [reply _data/training-modules.yaml:15]: That's the description for the next one :)

Comment (klieret) [reply _data/training-modules.yaml:15]: (I think a yaml formatter originally sorted the keys alphabetically)

Comment (wdconinc) [reply _data/training-modules.yaml:15]: Haha, I see now.

**Pull request merged**